### ISSUE TITLE
azure - reduce validation in tests

### ISSUE THREAD

**Issue by stefangordon** [cloud-custodian/cloud-custodian#4873]
There are a significant number of tests which do schema validation in the Azure tests.  Schema validation is absurdly slow, so we should remove validation from tests which are not explicitly testing validation.

**Comment by kapilt** [comment]
fwiw, schema validation on tests in other providers is only done in ci behind an environment variable, the measured performance latency was about 20%, mostly this is controlled in tests via validate parameter to load_policy which is only explicitly set in tests that are testing their validation methods.

**Comment by stefangordon** [comment]
Yep exactly - I noticed some test classes got carried away with copy/paste have have explicit validate on large numbers of tests.  In a profiled run of the Azure provider with validation disabled in the environment variable I still see 75% of the test time lost to schema validation.

**Comment by kapilt** [comment]
definitely worth turning off by default unless its specific to the test. also its worth noting that care is needed with profiling for performance timings, the instrumentation can pertub the results significantly due to the function stacks depths in validation, the overhead i noted previously was done sans profiling, just measuring the effect of turning on the validation env var. when we drop py2 support, there's some significant speedups to be had via changing validation frameworks (though requires revisting our semantic error handling).

**Comment by stefangordon** [comment]
Re drop py2: Are you referring to that `fastjsonschema` implementation? I tried it out and its very fast, but it doesn't have a huge user base.
